extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -381,11 +387,9 @@ public function getPage() {
 						$data['article'] = $articleData;
 
 						if ( !$title->isContentPage() ) {
-							// This does two things:
-							// - gets displayTitle from parser function {{DISPLAYTITLE:displayTitle}}
-							// - removes the namespace prefix from it
-							$data['article']['displayTitle'] =
-								Title::newFromText( $data['article']['displayTitle'] )->getText();
+							// Remove the namespace prefix from display title
+							$displayTitle = Title::newFromText( $displayTitle )->getText();
```

Comment:
It's not so easy with handling that exception. `newFromText()` throws `MWException` which we don't catch. We catch only `WikiaHttpException` which isn't ancestor for `MWException`.